frontend/lib/framework/error-classes-list.cpp

```diff
@@ -537,4 +537,138 @@ void ErrorProcDefExplicitAnonFormal::write(ErrorWriterBase& wr) const {
   wr.code(fn, {formal});
 }
 
+/* parser errors */
+
+void ErrorParsing::write(ErrorWriterBase& wr) const {
+  auto loc = std::get<const Location>(info);
+  auto errorMessage = std::get<std::string>(info);
+  assert(errorMessage.back() != '.' &&
+         "expected no period at end of ErrorParsing message");
+  errorMessage.push_back('.');
+  wr.heading(kind_, type_, loc, errorMessage);
+}
+
+void ErrorCannotAttachPragmas::write(ErrorWriterBase& wr) const {
+  auto loc = std::get<const Location>(info);
+  wr.heading(kind_, type_, loc, "cannot attach pragmas to this statement.");
+}
+
+void ErrorInvalidIndexExpr::write(ErrorWriterBase& wr) const {
+  auto loc = std::get<const Location>(info);
+  wr.heading(kind_, type_, loc, "invalid index expression.");
+}
+
+void ErrorRecordInheritanceNotSupported::write(ErrorWriterBase& wr) const {
+  auto loc = std::get<const Location>(info);
+  wr.heading(kind_, type_, loc,
```

Comment:
In other error message like this (e.g., "can't use owned with record", we do "Bla declared as a record here". This can even be a note, followed by code output.

Reply:
Same issue as above prevents outputting code, but I can give a note with a location.

Reply:
Adding code here now!